REFERENCE.md

```diff
@@ -481,6 +542,20 @@ an optional lambda to return a default value in case the lookup fails
 
 ## Data types
 
+### <a name="Dnsquery--Config_info"></a>`Dnsquery::Config_info`
+
+Type to validate the config_info passed to Resolve::DNS.new
+
+Alias of
+
+```puppet
+Struct[{
+  Optional['nameserver'] => Variant[Stdlib::IP::Address::Nosubnet, Array[Stdlib::IP::Address::Nosubnet]],
+  Optional['search']     => Array[Stdlib::Fqdn],
+  Optional['ndots']      => Integer[1,63],
```

Comment:
According to [the Resolv::DNS#new documentation](https://ruby-doc.org/stdlib-2.7.2/libdoc/resolv/rdoc/Resolv/DNS.html#method-c-new), when passing a hash it:

> Must contain `:nameserver`, `:search` and `:ndots` keys.

Looks like in fact it *SHALL* have a `:nameserver` and *MAY* have `:search` and `:ndots`.

```suggestion
  'nameserver'       => Variant[Stdlib::IP::Address::Nosubnet, Array[Stdlib::IP::Address::Nosubnet]],
  Optional['search'] => Array[Stdlib::Fqdn],
  Optional['ndots']  => Integer[1,63],
```